src/main/java/Parser.java

```diff
@@ -0,0 +1,446 @@
+package duke;
+
+import duke.DukeException;
+import duke.Command;
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.Task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.FormatStyle;
+import java.util.Optional;
+
+/**
+ * Parser deals with the full commands given by the user.
+ * Inputs given are deciphered and corresponding responses are outputted.
+ */
+
+public class Parser {
+	public static Command parse(String fullCommand) throws DukeException {
+		if(fullCommand.equals("list")) {
+			return list();
+		} else if (fullCommand.equals("bye")) {
+			return bye();
+		} else if (getWord(fullCommand).equals("done")) {
+			return done(fullCommand);
+		} else if (getWord(fullCommand).equals("todo")) {
+			return todo(fullCommand);
+		} else if (getWord(fullCommand).equals("deadline")) {
+			return deadline(fullCommand);
+		} else if (getWord(fullCommand).equals("event")) {
+			return event(fullCommand);
+		} else if (getWord(fullCommand).equals("delete")) {
+			return delete(fullCommand);
+		} else if(getWord(fullCommand).equals("save")) {
+			return save();
+		} else if (getWord(fullCommand).equals("find")) {
+			return find(fullCommand);
+		} else{
+			throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+		}
+	}
+
+	/**
+	 * Returns the first word in the string indicated by the first blank space.
+	 * If there are no blank space, the entire string is returned.
+	 *
+	 * @param string
+	 * @return
+	 */
+	public static String getWord(String string) {
+
+		int firstSpaceIndex = string.indexOf(' ');
+		if (firstSpaceIndex == -1) {
+			return string;
+		}
+		return string.substring(0, firstSpaceIndex);
+	}
+
+	/**
+	 * Returns an implemented instance of the Command interface when the list command is invoked.
+	 *
+	 * @return
+	 */
+	public static Command list() {
+		return new Command() {
+			@Override
+			public String execute(TaskList taskList, Ui ui, Storage storage) {
```

Comment:
Should this part be extracted out? Perhaps consider creating multiple command class to handle different commands in order to improve the readability of following command method? :)